pil-analyzer/src/pil_analyzer.rs

```diff
@@ -297,6 +297,21 @@ impl PILAnalyzer {
             }
         }
 
+        let struct_declarations: HashMap<String, StructDeclaration> = self
+            .definitions
+            .iter()
+            .filter_map(|(name, (_, value))| {
+                if let Some(FunctionValueDefinition::TypeDeclaration(TypeDeclaration::Struct(
+                    struct_decl,
+                ))) = value
+                {
+                    Some((name.clone(), struct_decl.clone()))
```

Comment:
Is it possible to avoid this clone? We are iterating oven the definitions below, is it possible to collect references to the struct declarations there?

Reply:
Wait, I think this is not the right way at all.